### DISCUSSION TITLE
[Merge branches] Not really a git merge?

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#34] [Q&A] [answered]
Hi, me and my team tried the new feature of the Mia-Platform console to merge 2 branches of the configurations.
The goal was to merge the branch `tickets-proxy-manager` into the branch `master` but we cautiously created a new branch for master, just to analyze the commit and to be sure. Following what we did:
- create the new branch `support-testing-merge` from `master`
- merge the branch `tickets-proxy-manager` into the branch `support-testing-merge` **from the Mia-Platform console**
- analyze the commit changes and the Git Graph
- merge the branch `support-testing-merge` into the branch `master` **from Gitlab**

Watching the Git Graph we noticed that the "Merge" feature is not really a _Git Merge_, because the destination branch doesn't have the commit history, like in the following screenshot:
<img width="1187" alt="Schermata 2022-09-27 alle 11 50 49" src="https://user-images.githubusercontent.com/28152586/192495631-aa1922ef-6cab-4210-9182-cfbce6420327.png">

Is this the wanted behavior?
I think that the user should know this important difference with the `git merge` feature.

**Comment by danibix95** [comment]
Could you please verify whether merged configuration are actually found on the `master` branch?

I believe it is a Gitlab issue, since I noticed the same behaviour in other repositories not managed directly by Console (e.g. services repositories). This randomly occurs after a merge request is actually merged into the main branch.

**Reply by fredmaggiowski** [reply]
Please note that the feature itself is not a `git merge` but a configuration merge, so I think this is not the case.

**Comment by davidebianchi** [comment, marked as answer]
The merge is not a git merge, but it's a merge between two different configurations. In a repo there is not only the configurations files, but also resources, extensions and other files.

In the merge, the extensions I suppose are not merged correctly inside the branch.
Such others type of files, like IngressRoute, Certificate, SleepInfo, ...
This is an issue, and sure we should notify it to the user!

**Reply by fredmaggiowski** [reply]
I confirm that currently we only support the resources you see in the merge modal sidebar; extensions and other files are not supported by the feature and must be manually migrated to the branch.

**Reply by frengooo** [reply]
Ok, now it's more clear to me, thank you guys.
I didn't know that the extensions files are not merged, I agree that we must add an explicit warning because it's easy to confuse this merge with a `git merge`, considering that the configurations of a project are "git centric".
The discussion can be closed for me.

**Reply by peppemanzi** [reply]
However this choice implies that you loose the commit history when merging from Console, since you usually want to delete the source branch after that, in order to have a clean repository. I think that this behavior is at odd with the versioning approach of the Console, that imho is one of its greatest advantages. Do we have any idea on how to use a git merge and make the user aware of the sideffects or (better) make him able to handle resources, extensions and other files in the repo?
